### ISSUE TITLE
Make registering ViewControllers more save

### ISSUE THREAD

**Issue by lightsprint09** [AliSoftware/Dip-UI#29]
Would it make sense to have an extension on `DependencyContainer ` like the following:

```swift
extension DependencyContainer {
  public func registerViewController<T: StoryboardInstantiatable>(_ scope: ComponentScope = .shared, type: T.Type = T.self, tag: DependencyTagConvertible? = nil, factory: @escaping () throws -> T) -> Dip.Definition<T, ()> {
    return register(scope, type: type, tag: tag, factory: factory)
  }
}
```

This would help to not forget to conform to `StoryboardInstantiatable `.

**Comment by ilyapuchka** [comment]
It might be, but nothing will stop you from still using `register` though, that will not warn you. I think it would be better just to add more specific errors when resolving of subclasses on `UIViewController` fails (I believe we can get subclasses chain from Mirror of type)

**Comment by lightsprint09** [comment]
I see, it makes sense to improve runtime error. But if most developers go with the convention of using `registerViewController `, they would win compile time errors which could be powerful as well.

**Comment by ilyapuchka** [comment]
Closed with #31
